src/actions/braze/braze.ts

```diff
@@ -0,0 +1,172 @@
+      throw "Missing primary Braze key."
+    }
+
+    const exportValue = request.formParams.braze_segment  || String(BrazeConfig.EXPORT_DEFAULT_VALUE)
+    const brazeAttribute = { add : [ exportValue ] }
+
+    let totalCnt = 0
+    let fieldlist: Hub.Field[] = []
+    let bzIdField = ""
+    let rows: any[] = []
+    try {
+      await request.streamJsonDetail({
+        onFields: (fields) => {
+          fieldlist = Hub.allFields(fields)
+          for (const field of fieldlist) {
+            if (field.tags  && field.tags.find((tag: string) => tag === BrazeConfig.BRAZE_ID_TAG )) {
+              bzIdField = field.name
+            }
+          }
+          if (!bzIdField) {
+            throw "Primary Braze key not found."
+          }
+        },
+        onRow: (row) => {
+          if (totalCnt < BrazeConfig.MAX_EXPORT) {
+            const entry: { [key: string]: any } = {}
+            entry[String(request.formParams.braze_key)] = row[bzIdField].value
+            entry[String(BrazeConfig.LOOKER_ATTRIBUTE_NAME)] = brazeAttribute
+            // Only update existing records to prevent unknown data sources
+            entry._update_existing_only = true
+            rows.push(entry)
+
+            totalCnt++
+            if (rows.length === BrazeConfig.MAX_LINES) {
+              this.sendChunk(urlendpoint, request.params.braze_api_key, rows)
+                .catch( (e) => {
+                  return new Hub.ActionResponse({success: false, message: e.message })
+                })
+              rows = []
+            }
+          } else if (rows.length > 0) {
+            this.sendChunk(urlendpoint, request.params.braze_api_key, rows)
+              .catch( (e) => {
+                return new Hub.ActionResponse({success: false, message: e.message })
+              })
+            rows = []
+          }
+        },
+      })
+
+      if (rows.length > 0) {
+        this.sendChunk(urlendpoint, request.params.braze_api_key, rows)
+          .catch( (e) => {
+            return new Hub.ActionResponse({success: false, message: e.message })
+          })
+        rows = []
+      }
+    } catch (e) {
+      return new Hub.ActionResponse({success: false, message: e.message })
+    }
+    return new Hub.ActionResponse({success: true, message: "ok"})
+  }
+  async form() {
+    const form = new Hub.ActionForm()
+    form.fields = [{
+      label: "Unique Key",
+      name: "braze_key",
+      description: " Primay key for user to map to within Braze.",
+      required: true,
+      options: [
+        {name: "external_id", label: "external_id"},
+        {name: "braze_id", label: "braze_id"},
+      ],
+      type: "select",
+      default: "external_id",
+    }, {
+      label: "Export Label",
+      name: "braze_segment",
+      description: "Name of export (Appends to Custom Attribute Array '" +
+        BrazeConfig.LOOKER_ATTRIBUTE_NAME + "'). Defaults to '" + BrazeConfig.EXPORT_DEFAULT_VALUE + "'." ,
```

Comment:
You should be able to provide a default value by specifying `default` for a form field.

Reply:
removed thanks

Reply:
added thanks